primitives/primproc/bppsendthread.h

```diff
@@ -100,7 +110,11 @@ class BPPSendThread
     bool waiting;
     boost::mutex ackLock;
     boost::condition okToSend;
-
+    // Condition to prevent run away queue
+    bool respondWait;
+    boost::mutex respondLock;
```

Comment:
I know you hate mutexes. Not fond of them myself. But I need flow control between BPP and SendThread. Sendthread already has flow control between it and ExeMgr. So it can back up depending on how fast ExeMgr can consume. When it backs up to much, Sendthread queues up until OOM. 
So, in the normal small query case, send thread is not utilized. It doesn't come into play until ExeMgr triggers flowcontrol which happens when it's receive buffer backs up.

Once ExeMgr initiates flow control, all further calls by BPP to sendResponse() goes to the Queue rather than directly to TCP and hence to ExeMgr.

BPPSendThread doesn't use flow control between itself and BPP until it also backs up -- if (sizeTooBig();

So, while we are introducing a mutex, it doesn't affect other queries, and it only affects the current query iff it gets too backed up. Most queries will never notice.

There is a side effect. In the past, BPP flow was restricted in BatchPrimitiveProcessor::operator()()
```
            if (!sendThread->okToProceed())
            {
                freeLargeBuffers();
                return -1; // the reschedule error code
            }

```
This freed up the BPP instance to continue on another jobstep. But if the new flow control kicks in, the BPP will be blocked and not released to work another job step. I don't believe this is an issue as another BPP instance will be available.